pkg/controller/statefulset/stateful_pod_control.go

```diff
@@ -192,6 +193,23 @@ func (spc *realStatefulPodControl) createPersistentVolumeClaims(set *apps.Statef
 		case err != nil:
 			errs = append(errs, fmt.Errorf("Failed to retrieve PVC %s: %s", claim.Name, err))
 			spc.recordClaimEvent("create", set, pod, &claim, err)
+		case err == nil:
+			if pvc.DeletionTimestamp != nil {
+				// double check if the pvc is in actually terminating state.
+				// if get pvc failed collect it;
+				// if not found or in terminating also collect error;
+				// if found pvc and not in terminating state, do not collect it.
+				actualPVC, getErr := spc.client.CoreV1().PersistentVolumeClaims(claim.Namespace).Get(claim.Name,
```

Comment:
I don't see why rechecking the deletionTimestamp by live GET helps here. The same object may not transition back and you get race on detecting the recreate case where you can miss it so we'd need a reconciliation logic for that case anyways.

Watching if pods reference deleted PVC and reconciling it as @kow3ns suggested in https://github.com/kubernetes/kubernetes/pull/79164#issuecomment-509414704 sounds more reliable

Reply:
> I don't see why rechecking the deletionTimestamp by live GET helps here. The same object may not transition back and you get race on detecting the recreate case where you can miss it so we'd need a reconciliation logic for that case anyways.
> 
> Watching if pods reference deleted PVC and reconciling it as @kow3ns suggested in [#79164 (comment)](https://github.com/kubernetes/kubernetes/pull/79164#issuecomment-509414704) sounds more reliable

Thanks, I would do that later